disco_aws_automation/disco_deploy.py

```
@@ -390,6 +391,103 @@ def handle_tested_ami(self, old_hostclass_dict, ami, desired_size,
+                    (not run_tests or self.run_integration_tests(ami, wait_for_elb=uses_elb))):
+                # If testing passed, mark AMI as tested
+                self._promote_ami(ami, "tested")
+                # Get list of instances in group
+                group_instance_ids = [inst.instance_id for inst in
+                                      self._disco_autoscale.get_instances(group_name=new_group.name)]
+                group_instances = self._disco_aws.instances(instance_ids=group_instance_ids)
+                # If we are actually deploying and are able to leave testing mode and start serving requests
+                if deployable and self._set_testing_mode(hostclass, group_instances, False):
+                    logging.info("Successfully left testing mode for group %s", new_group.name)
+                    # Update ASG to exit testing mode and attach to the normal ELB if applicable.
+                    self._disco_aws.spinup([new_group_config], group_name=new_group.name)
+                    if uses_elb:
+                        try:
+                            # Wait until the new ASG is registered and marked as healthy by ELB.
+                            self._disco_elb.wait_for_instance_health_state(hostclass=hostclass,
+                                                                           instance_ids=group_instance_ids)
+                        except TimeoutError:
+                            logging.exception("Waiting for health of instances attached to ELB timed out")
+                            # Destroy the testing ASG
+                            self._disco_autoscale.delete_groups(group_name=new_group.name, force=True)
+                            if uses_elb:
+                                # Destroy the testing ELB
+                                self._disco_elb.delete_elb(hostclass, testing=True)
+                            return
```

we should return an error status of some sort and propagate that up so that we can decide whether we want to exit as 0 or 1